src/custom/utils/gas.ts

```diff
@@ -0,0 +1,37 @@
+import { TransactionResponse } from '@ethersproject/providers'
+import { GASNOW_PRICE_ENDPOINT } from 'constants/index'
+import { Overrides } from 'ethers'
+import { parseUnits } from 'ethers/lib/utils'
+
+interface PriceEstimate {
+  rapid: number
+  fast: number
+  slow: number
+  standard: number
+  timestamp?: number
+}
+
+export async function getGasPriceEstimate(): Promise<PriceEstimate | void> {
+  try {
+    const json = await fetch(GASNOW_PRICE_ENDPOINT)
+    const estimate = await json.json()
+
+    if (estimate.code === 200) {
+      return estimate.data
+    }
+  } catch (err) {
+    console.log(`Gas estimation fetch failed`, err)
+    return
+  }
+}
+
+export async function applyCustomGasPrice(fn: (options: Overrides) => TransactionResponse | Promise<any>) {
```

Comment:
With that name (`applyCustomGasPrice`) I would expect the signature to be:

`applyCustomGasPrice(fn: (options: Overrides) => Overrides)` wouldn't that be simpler? 

Also, if `options` has gas estimation, i wouldn't call to `gasPriceData = await getGasPriceEstimate()` unless it's undefined. This way, we just provide the gas price if non is specified in the params